cobalt/updater/network_fetcher.cc

```diff
@@ -0,0 +1,231 @@
+  // `Content-Type` header present in the |ResourceRequest| above.
+  simple_url_loader_->AttachStringForUpload(post_data, content_type);
+  simple_url_loader_->SetOnResponseStartedCallback(base::BindOnce(
+      &NetworkFetcher::OnResponseStartedCallback, base::Unretained(this),
+      std::move(response_started_callback)));
+  simple_url_loader_->SetOnDownloadProgressCallback(base::BindRepeating(
+      &NetworkFetcher::OnProgressCallback, base::Unretained(this),
+      std::move(progress_callback)));
+  constexpr size_t kMaxResponseSize = 1024 * 1024;
+  simple_url_loader_->DownloadToString(
+      url_loader_factory_.get(),
+      base::BindOnce(
+          [](const network::SimpleURLLoader* simple_url_loader,
+             PostRequestCompleteCallback post_request_complete_callback,
+             std::unique_ptr<std::string> response_body) {
+            LOG(INFO) << "post_request_complete_callback, response_body="
+                      << response_body->c_str();
+            std::move(post_request_complete_callback)
+                .Run(std::move(response_body), simple_url_loader->NetError(),
+                     GetStringHeader(simple_url_loader, kHeaderEtag),
+                     GetStringHeader(simple_url_loader, kHeaderXCupServerProof),
+                     GetInt64Header(simple_url_loader, kHeaderXRetryAfter));
+          },
+          simple_url_loader_.get(), std::move(post_request_complete_callback)),
+      kMaxResponseSize);
+}
+
+#if defined(IN_MEMORY_UPDATES)
+void NetworkFetcher::DownloadToString(
+    const GURL& url,
+    std::string* dst,
+    ResponseStartedCallback response_started_callback,
+    ProgressCallback progress_callback,
+    DownloadToStringCompleteCallback download_to_string_complete_callback) {
+  LOG(INFO) << "NetworkFetcher::DownloadToString, url = " << url;
+  // TODO(b/444006168): implement this to support in-memory updates
+}
+
+#else
+void NetworkFetcher::DownloadToFile(
+    const GURL& url,
+    const base::FilePath& file_path,
+    ResponseStartedCallback response_started_callback,
+    ProgressCallback progress_callback,
+    DownloadToFileCompleteCallback download_to_file_complete_callback) {
+  LOG(INFO) << "NetworkFetcher::DownloadToFile, file_path = "
+            << file_path.value().c_str() << ", url = " << url;
+  CHECK(!simple_url_loader_);
+  auto resource_request = std::make_unique<network::ResourceRequest>();
+  resource_request->url = url;
+  resource_request->method = "GET";
+  resource_request->load_flags = net::LOAD_DISABLE_CACHE;
+  simple_url_loader_ = network::SimpleURLLoader::Create(
+      std::move(resource_request), traffic_annotation);
+  simple_url_loader_->SetRetryOptions(
+      kMaxRetriesOnNetworkChange,
+      network::SimpleURLLoader::RetryMode::RETRY_ON_NETWORK_CHANGE);
+  simple_url_loader_->SetAllowPartialResults(true);
+  simple_url_loader_->SetOnResponseStartedCallback(base::BindOnce(
+      &NetworkFetcher::OnResponseStartedCallback, base::Unretained(this),
```

Comment:
nit: Use of `base::Unretained()` needs to be documented.
I see for example `::Cancel()` below `reset()`s this 
`simple_url_loader_`, does that mean we would never
get calls back after `simple_url_loader_` is destroyed?